amaranth-chrome-ext/src/CalorieLabeller.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * An object which labels a dish name as low, average, or high-calorie.
+ */
+class CalorieLabeller {
+  /**
+   * Creates a CalorieLabeller object.
+   * @constructor
+   * @param {Map<string, number>} tokenizer A mapping from dish name tokens to
+   * unique integers
+   * @param {tf.LayersModel} model Tensorflow.js layers ML model
+   */
+  constructor(tokenizer, model) {
+    /** @private @const @type {Map<string, number>} */
+    this.tokenizer_ = tokenizer;
+    /** @private @const @type {tf.LayersModel} */
+    this.model_ = model;
+  }
+
+  /**
+   * Labels a single dish as high or low calorie.
+   * @param {string} dishName The name of the dish to label
+   * @return {CalorieLabel} The calorie label for the dish named dishName
+   */
+  label(dishName) {
```

Comment:
Consider breaking this function body up into sub functions if you want more code readability.

Reply:
I simplified some of the logic in https://github.com/googleinterns/amaranth/pull/44/commits/1c9024796bc63e0ee624c795f5ab16a12bc1434d. That plus the removal of unnecessary comments in https://github.com/googleinterns/amaranth/pull/44/commits/f647285d93ec9d9fbeb789cf6e6e452b880fb0b1 makes it sufficiently readable, in my opinion